collectors/metrics/cmd/metrics-collector/main.go

```diff
@@ -301,16 +299,36 @@ func (o *Options) Run() error {
 	// Some packages still use default Register. Replace to have those metrics.
 	prometheus.DefaultRegisterer = metricsReg
 
-	err, cfg := initConfig(o)
+	cfgRR, err := initShardedConfigs(o, AgentRecordingRule)
```

Comment:
There is probably some code we can extract in a specific function from `initShardedConfigs`. It mutates the options objects, doing the same processing 3 times.

Reply:
Yeah you're right, just didn't want to change even more things on sensitive code path